### PR TITLE
2.3.1 Documentation Update

### PR DESCRIPTION
**Pull request by Valenzione**
Here is the list of changes in the docs:
1. Updated serving code in the `Tutorials` and `Getting Started` sections. Now these serving snippets returns `Dict[str, np.array]` instead of `PredictResponse`. This reduces LOC & complexity of these snippets.
2. Update the `What is Hydrosphere` section. Added more information and graphics, emphasized Why the end user may need Hydrosphere, and What he should not expect from it. 
3. Revamped the `Concepts` section a bit. Added a few concepts about Models' Signature, described Metrics, removed deprecated concepts of Batch Metrics.
4. Changed README.md to exclude features we do not offer yet
5. Added all sorts of information related to `hydrosdk` - Installation, Examples, etc.
6. Added How-to recipe for using private PyPI server
7. Added the new section dedicated to all Hydrosphere services, including those behind the closed license
8. Updated the Roadmap

### CODE REVIEW

File: docs/src/main/paradox/services/serving/manager.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@@@ note
+Manager is an open-source service available [here](https://github.com/Hydrospheredata/hydro-serving-manager)
+as a part of [hydro-serving](https://github.com/Hydrospheredata/hydro-serving) project.
+@@@
+
+Manager is responsible for:   
+
+* Building a Docker Image from your ML model for future deployment
+* Storing these images inside a Docker Registry deployed alongisde with manager service
```
Comment (tidylobster) [inline docs/src/main/paradox/services/serving/manager.md:11]:
```suggestion
* Storing these images inside a configured Docker Registry or one, deployed alongisde with the Hydrosphere platform.  
```

File: docs/src/main/paradox/tutorials/serving/python.md [hunk 2]
```diff
@@ -26,15 +30,19 @@ $ touch src/func_main.py
...
+Generally, we use `hydrosphere/serving-runtime-python-3.7` runtime for serving Python models.
+ This runtime uses `src/func_main.py` script as an entry point.
+  You may create any arbitrary Python application within your model,
+   just keep in mind that the entry point of your script has to be in `src/func_main.py`.
 @@@
 
 ## Model dependencies
 
-By default the `hydrosphere/serving-runtime-python-3.6` runtime does not have any scientific packages pre-installed, you will have to manage this yourself. Let's add `requirements.txt`:
+By default the `hydrosphere/serving-runtime-python-3.7` runtime does not have any
```
Comment (tidylobster) [inline docs/src/main/paradox/tutorials/serving/python.md:41]: We now depend on numpy (through sdk), so we can state it here.

File: docs/src/main/paradox/tutorials/serving/python.md [hunk 3]
```diff
@@ -87,8 +93,10 @@ contract:  # time to remember your Python function signature
 If `.contract.name` is not set, it defaults to `predict`.
 @@@
 
-This file describes the model, its name, type, payload, and service contract. Contract declares input and output fields for the model; data types, shapes, and profile information. Profile information is used to calculate profiles of your training/production data.
-
+'serving.yaml' file describes the model, its name, type, payload, and contract.
+ The model contract declares input and output fields' data types and shapes.
+  Fields` profile information is used to tell Hydrosphere how to interpret this field.
```
Comment (tidylobster) [inline docs/src/main/paradox/tutorials/serving/python.md:98]:
```suggestion
  Fields' profile information is used to tell Hydrosphere how to interpret this field.
```

**Comment by tidylobster** [comment]
I've noted a few minor issues, but overall this is a great work!

File: docs/src/main/paradox/install/sdk.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import pandas as pd
+
+cluster = Cluster("http://localhost", grpc_address="localhost:9090")
+
+app = Application.find(cluster, "my-model")
+predictor = app.predictor()
+
+df = pd.read_csv("path/to/data.csv")
+for row in df.itertuples(index=False):
+    predictor.predict(row)
```
[the remark this answers is not in the document]
Comment (Valenzione) [reply docs/src/main/paradox/install/sdk.md:32]: It'll work since we've added support for `namedtuple` :)

**Comment by Valenzione** [comment]
Closing, changes from this PR were implemented in #328

**Pull request closed without merging**